spec/app_store/verifier_spec.rb

```diff
@@ -80,6 +81,27 @@
     end
   end
 
+  describe 'subscription' do
+    let(:environment) { :production }
+
+    it 'uses production endpoint without retry on success' do
+      with_mocked_verifier(receipt_collection) do
+        subject.verify_subscription(
+          data, secret
+        ).must_be_same_as receipt_collection
+        assert_recorded([production_endpoint, data, secret])
+      end
+    end
+
+    it 'only uses production endpoint for normal failures' do
+      failure = get_failure(21_000)
+      with_mocked_verifier(failure) do
+        subject.verify_subscription(data, secret).must_be_same_as failure
+        assert_recorded([production_endpoint, data, secret])
+      end
+    end
+  end
+
```

Comment:
So verifying subscriptions works basically like standard purchases for iOS? (Except the part around the shared secret)